### PR TITLE
feature/18

### PR DESCRIPTION
**Pull request by bartimys1**

### CODE REVIEW

File: HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/EngGame.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                   EngArr[i] *= a;
+                NumbStory.Push(a);
+                GameStory.Push(LeftLevel);              
+                Print();
+            }
+            else if (a == 0)
+            {
+                try
+                {
+                    GameStory.Pop().Invoke();
```
Comment (alch-0) [inline HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/EngGame.cs]: Unless I am missing something it seems that  LeftLevel() method can be invoked directly, eliminating the need for Stack<Action> GameStory variable in EndGame class along with calling GameStory.Push(LeftLevel) inside of Play method.

**Comment by alch-0** [comment]
After checking other implementations of this task I think that perhaps there has to be the following check as well:
the number that the user inputs has to be accepted only in case such a number is present among the numbers last printed (meaning that the user can't just enter any random number).
The task description doesn't clearly indicate this requirement so, I guess, this needs further clarification.

File: HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/EngGame.cs (hunk 1 above)
Comment (bartimys1) [reply HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/EngGame.cs]: Initially, I wanted to use the stack<action<int>>, but it works differently. As a result, I have such a crutch

**Comment by bartimys1** [comment]
#18

File: HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/EngGame.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+using System.Collections.Generic;
+
+namespace GameWithNumbers
+{
+    public class EngGame
+    {
+        
+        Random rnd = new Random();
+        private int[] EngArr = new int[5];
+        Stack<Action> GameStory = new Stack<Action>();
```
Comment (AVLozhechkin) [inline HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/EngGame.cs]: Im not sure that we should save level's story. 
The idea is that on each level we can go back by dividing current level on current chosen number and can go to the next level by multiplying, so there is no reason to store this data.

Comment (bartimys1) [reply HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/EngGame.cs]: Thank you for efficiency)

File: HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/Program.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            EngGame LetsPlay = new EngGame();
+
+            while (true)
+            {
+                try
+                {
```
Comment (AVLozhechkin) [inline HW1/Ivan_Pavlov/GameWithNumbers/GameWithNumbers/Program.cs]: To improve UI i would add here something like 
Console.Write("\nSelect a number ");

**Pull request merged**